kafka-connect-s3/src/test/java/io/confluent/connect/s3/TopicPartitionWriterTest.java

```diff
@@ -1079,6 +1090,46 @@ public void testRecordKeysAndHeadersWritten() throws Exception {
     verifyRecordElement(expectedKeyFiles, 3, sinkRecords, RecordElement.KEYS);
   }
 
+  @Test
+  public void testRecordHeadersWrittenJson() throws Exception {
```

Comment:
tests look good, but looks like there is a good opportunity for parametrizing these tests across all of the formats, might be worth looking into but not necessary for this PR